sql/src/main/java/io/crate/operation/auth/AuthenticationProvider.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+
+import javax.annotation.Nullable;
+import java.net.InetAddress;
+import java.util.Iterator;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+
+@Singleton
+public class AuthenticationProvider {
+
+    private Authentication authService;
+    @VisibleForTesting
+    public static final Authentication NOOP_AUTH = new Authentication() {
+        @Override
+        public boolean enabled() {
+            return false;
```

Comment:
no, we still need the `enabled` because it is possible that the auth module is loaded, but disabled. (in case the `auth.host_based` setting is not present / the hbaConf map is `null`).